spec/unit/functions/dig_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'spec_helper'
+
+require 'puppet_spec/compiler'
+require 'matchers/resource'
+
+describe 'the dig function' do
+  include PuppetSpec::Compiler
+  include Matchers::Resource
+
+  it 'returns a value from an array index via integer index' do
+    expect(compile_to_catalog("notify { [testing].dig(0): }")).to have_resource('Notify[testing]')
+  end
+
+  it 'returns undef if given an undef key' do
+  expect(compile_to_catalog(<<-SOURCE)).to have_resource('Notify[testing]')
+    notify { "test${[testing].dig(undef)}ing": }
```

Comment:
Nitpick, but you don't know if the return is really an undef or something else that represents itself as an empty string (applies to several tests).